plc4j/tools/connection-cache/src/main/java/org/apache/plc4x/java/utils/cache/CachedPlcConnectionManager.java

```diff
@@ -58,21 +58,33 @@ public CachedPlcConnectionManager(PlcConnectionManager connectionManager, Durati
         this.connectionContainers = new HashMap<>();
     }
 
+    @Override
     public PlcConnection getConnection(String url) throws PlcConnectionException {
+        return getConnection(url,null);
+    }
+
+    @Override
+    public PlcConnection getConnection(String url, PlcAuthentication authentication) throws PlcConnectionException {
         ConnectionContainer connectionContainer;
         synchronized (connectionContainers) {
             connectionContainer = connectionContainers.get(url);
```

Comment:
As this now includes the authentication, we should probably have a more generic "getContainerId" method, that takes a url and authentication and if authentication is not null, then it sort of does a "url + authentication.toString()" to allow multiple connections with different users (I could imagine this being a topic in OPC-UA)

Reply:
I never use authentication. Just follow the api. I can remove this change

Reply:
Well then we should leave it as it was ... either we support it and then we should really support it, or we don't and we throw an exception (as it was before)